server/rooms.js

```diff
@@ -1693,12 +1694,18 @@ let Rooms = Object.assign(getRoom, {
 				user.inviteOnlyNextBattle = false;
 			}
 		}
-		if (options.tour && !room.tour.modjoin) inviteOnly = [];
 		if (inviteOnly.length) {
-			room.modjoin = '%';
-			room.isPrivate = 'hidden';
-			room.privacySetter = new Set(inviteOnly);
-			room.add(`|raw|<div class="broadcast-red"><strong>This battle is invite-only!</strong><br />Users must be invited with <code>/invite</code> (or be staff) to join</div>`);
+			const prefix = battle.forcedPublic();
+			if (prefix) {
+				room.isPrivate = false;
+				room.modjoin = null;
+				room.add(`|raw|<div class="broadcast-red"><strong>This battle is required to be public due to a player having a name prefixed by '${prefix}'.</div>`);
```

Comment:
I would color this non-red so it's clear it's not private. Maybe like blue?

Reply:
```suggestion
				room.add(`|raw|<div class="broadcast-blue"><strong>This battle is required to be public due to a player having a name prefixed by '${prefix}'.</div>`);
```

suggestion for one-click application, but feel free to decline this one

Reply:
I thought not being private when a user intended it to be is more critical to call out than it actually being private, which is why i went with red. But good point, if private already === red, then a user could just see the color and assume it means private. Blue is fine for me, even if its less scary/abnormal